internal/util/regexp.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2021-2023 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package util
+
+import (
+	"fmt"
+	"regexp"
+	"sync"
+)
+
+type RegexpCache struct {
+	cache map[string]*regexp.Regexp
+	mu    *sync.Mutex
+}
+
+func NewRegexpCache() *RegexpCache {
+	return &RegexpCache{
+		cache: make(map[string]*regexp.Regexp),
+		mu:    &sync.Mutex{},
+	}
+}
+
+// GetCompiledExpr lazily compiles (and stores) regexp.
+func (c *RegexpCache) GetCompiledExpr(re string) (*regexp.Regexp, error) {
+	if c.cache == nil {
+		c.cache = make(map[string]*regexp.Regexp)
+	}
```

Comment:
Can the `cache` be nil? It's initialised in the `NewRegexpCache` function, isn't it?

If it does need to be re-initalised, shouldn't it happen inside the lock?

Reply:
Yeah it can't be. This was one of my OCD just-in-case things, which in hindsight is just misleading. I'll remove